test/unit/util/GuardedStream.test.ts

```diff
@@ -26,11 +34,8 @@ describe('GuardedStream', (): void => {
       expect(isGuarded(stream)).toBe(true);
       expect(isGuarded(guarded)).toBe(true);
 
-      const listen = new Promise((resolve, reject): void => {
-        stream.on('end', resolve);
-        stream.on('error', reject);
-      });
-      await expect(readableToString(stream)).resolves.toBe('data');
+      const listen = getListenPromise(guarded);
```

Comment:
Not sure how useful this thing is here. `readableToString` already tests the same?

Reply:
Makes sense, I didn't pay that much attention to the older tests and just replaces these parts for simplicity. I might get back to you should I remember why this was a thing.